File: Algorithms/C++/01822-Sign of the Product of an Array.cpp
class Solution {
public:
    int arraySign(vector<int>& nums) {
        int sign = true;    // true is positive, false is negative

        for(const auto& num : nums){
            if(num == 0) return 0;

            if(num < 0) sign = !sign;
        }
        if(sign) return 1;
        return -1;

    }
};

/*
    tc: O(n)
    sc: O(1)

    boolean flag to track postive or negative
    if we see a zero, return 0




*/

File: Algorithms/Python3/01822-Sign of the Product of an Array.py
class Solution:
    def arraySign(self, nums: List[int]) -> int:
        if 0 in nums:
            return 0
        
        result = 1
        for num in nums:
            if num < 0:
                result *= -1
    
        return result
        
'''
    - find out if there is a zero in our nums (return 0)
    - set initial answer to 1
    - swap sign when we see a negative num
    - return answer


'''